### PR TITLE
Remove exponential backoff code

### PR DESCRIPTION
**Pull request by PettitWesley**
*Issue #, if available:*
https://github.com/aws/amazon-kinesis-firehose-for-fluent-bit/issues/23
https://github.com/fluent/fluent-bit/issues/2159

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by hencrice** [comment]
Any unit tests that need to be updated?

**Comment by sonofachamp** [comment]
Do we need to return an `FLB_RETRY` in place of the previous calls to backoff functionality?

**Comment by zackwine** [comment]
You'll need to update the `kinesis_test.go` file as well:  https://github.com/aws/amazon-kinesis-streams-for-fluent-bit/blob/master/kinesis/kinesis_test.go#L43

**Comment by PettitWesley** [comment]
@zackwine @hencrice Good point... I ran the integ tests but forgot to check the unit tests

@sonofachamp Yeah, you're right.

File: kinesis/kinesis.go [hunk 1]
```diff
@@ -199,11 +197,10 @@ func (outputPlugin *OutputPlugin) AddRecord(record map[interface{}]interface{},
 	newRecordSize := len(data) + len(partitionKey)
 
 	if len(outputPlugin.records) == maximumRecordsPerPut || (outputPlugin.dataLength+newRecordSize) > maximumPutRecordBatchSize {
-		err = outputPlugin.sendCurrentBatch()
+		retCode, err := outputPlugin.sendCurrentBatch()
 		if err != nil {
```
Comment (PettitWesley) [inline kinesis/kinesis.go:201]: There's a case in `processAPIResponse` where FLB_RETRY is returned but the error is nil; need to change this if block to account for that, or change that code

**Pull request merged**